client/state/data-layer/wpcom/sites/comments/replies/new/index.js

```diff
@@ -4,20 +4,29 @@
 import { COMMENTS_WRITE } from 'state/action-types';
 import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
 import {
+	hasParentCommentId,
 	dispatchNewCommentRequest,
 	updatePlaceholderComment,
 	handleWriteCommentFailure,
 } from 'state/data-layer/wpcom/sites/utils';
 
 export const writePostComment = ( { dispatch }, action ) => {
-	if ( ! action.parentCommentId ) {
+	if ( ! hasParentCommentId( action ) ) {
 		return;
 	}
 
 	const path = `/sites/${ action.siteId }/comments/${ action.parentCommentId }/replies/new`;
 	dispatchNewCommentRequest( dispatch, action, path );
 };
 
+const onSuccess = ( { dispatch }, action, ...args ) => ! hasParentCommentId( action )
+	? updatePlaceholderComment( { dispatch }, action, ...args )
+	: null;
+
+const onFailure = ( { dispatch }, action, ...args ) => ! hasParentCommentId( action )
+	? handleWriteCommentFailure( { dispatch }, action, ...args )
+	: null;
+
 export default {
-	[ COMMENTS_WRITE ]: [ dispatchRequest( writePostComment, updatePlaceholderComment, handleWriteCommentFailure ) ]
+	[ COMMENTS_WRITE ]: [ dispatchRequest( writePostComment, onSuccess, onFailure ) ]
```

Comment:
since WordPress has decided that replying to a post is different than replying to a comment (who knows why? anyone?) then it's fair to do that in Calypso as well.

> We might be able to make the dispatchRequest helper a bit smarter too, though that might require some

why would we want to? what inspection are you looking for? we don't have to use `dispatchRequest()`